game-core/src/main/java/games/strategy/engine/data/UnitTypeList.java

```diff
@@ -40,6 +44,22 @@ public UnitType getUnitType(final String name) {
     return types;
   }
 
+  /**
+   * Returns the unit support rules for the unit types. Computed once and cached afterwards.
+   *
+   * @return The unit support rules.
+   */
+  public Set<UnitSupportAttachment> getSupportRules() {
+    if (supportRules == null) {
+      supportRules =
+          UnitSupportAttachment.get(getData())
+              .parallelStream()
```

Comment:
Did you happen to measure if there is a performance boost of using parallel stream here instead of stream? `parallel` stream is oddly often slower than stream.

Reply:
I didn't try it, but I don't think it would make a big difference. The main issue was this was being called *a lot* and this change makes it get called just once, so it goes from taking a long time to taking no time at all (the remaining work getSortedSupport() is the result of the other calls).

Reply:
I'd recommend giving it a quick measure if you can. If not, blindly replacing with a standard stream.

Small tasks run in parallel often are actually much slower, it's surprising (I think the reasoning is the overhead for thread coordination is a lot, streams are not yet very optimized by the JDK, meanwhile single-threaded iteration is highly optimized still). Essentially it's to the point that one should not use parallel stream unless it has been benchmarked.

Thoughts?

Reply:
Tried it out and using parallelStream() is indeed slower (about 3x slower - 0.9ms vs. 0.3ms).